devito/passes/clusters/factorization.py

```diff
@@ -96,7 +110,7 @@ def collect_special(expr):
     # Collect common temporaries (r0, r1, ...)
     w_coeffs = Add(*w_coeffs, evaluate=False)
     symbols = retrieve_symbols(w_coeffs)
-    if symbols:
+    if len(set(symbols)) != len(symbols):
```

Comment:
retrieve_symbols should return a set already why is this needed?

Reply:
no, it returns a list: https://github.com/devitocodes/devito/blob/master/devito/symbolics/search.py#L157
since it's "mode=all"

we indeed want a list here because it's only when the same symbol appears twice (e.g. `[r42 .... r42 ...`) that we know it's worth attempting a factorization